client/lib/api/user_preferences.dart

```diff
@@ -59,15 +58,39 @@ class UserPreferences {
         .setBool(UserPreferenceKey.OnboardingCompletedV1.toString(), value);
   }
 
-  Future<bool> getTermsOfServiceCompleted() async {
+  /// Was the user shown the introductory pages as part of onboarding, using updated key for V1 launch
+  Future<bool> getOnboardingCompletedV2() async {
+    return (await SharedPreferences.getInstance())
+            .getBool(UserPreferenceKey.OnboardingCompletedV2.toString()) ??
+        false;
+  }
+
+  /// Was the user shown the introductory pages as part of onboarding, using updated key for V1 launch
+  Future<bool> setOnboardingCompletedV2(bool value) async {
+    return (await SharedPreferences.getInstance())
+        .setBool(UserPreferenceKey.OnboardingCompletedV2.toString(), value);
+  }
+
+  Future<bool> getLegalCompleted() async {
     return (await SharedPreferences.getInstance())
-            .getBool(UserPreferenceKey.TermsOfServiceCompleted.toString()) ??
+            .getBool(UserPreferenceKey.LegalCompleted.toString()) ??
         false;
   }
 
-  Future<bool> setTermsOfServiceCompleted(bool value) async {
+  Future<bool> setLegalAccepted() async {
+    var now = DateTime.now().millisecondsSinceEpoch;
```

Comment:
Tbh its unclear what use a local time is for definitive purposes.  You'd have to get a trusted timestamp say from a server for it to be meaningful.

Reply:
More specifically, it seems by a new approach you want to use this to determine whether to re-tos ppl in the future and a local time cannot tell you which tos/pn was accepted.  You could for example, associate all re-ToS/PNs with a new build version and store the app's build number at time of acceptance.  Or you could pull down the ToS/PN via HTTPS and store their combined hash or something.  Not legal advice.

Reply:
I see your point. This whole system needs to be reworked and doing it right is beyond what we can do in time for the launch.